internal/faketime/faketime.go

```diff
@@ -30,3 +30,36 @@ func AutoAdvance(t time.Time, dt time.Duration) func() time.Time {
 		return ret
 	}
 }
+
+// TimeAdvance allows controlling the passage of time. Intended to be used in
+// tests.
+type TimeAdvance struct {
+	mu sync.Mutex
+	t  time.Time
```

Comment:
nit (optional): this could be int64 and use atomic.AddInt64() to advance without a mutex.

Reply:
Then the type could not be `time.Time`.

We could have something like what's shown below, is that what you had in mind.

Not sure it makes much of a difference in tests though.